### DISCUSSION TITLE
Improving the "Performance points" calculation method

### DISCUSSION THREAD

**Discussion by nana123126** [ppy/osu#23348] [Ideas]
I think the "pp" (Performance points) system should have changed. The current "pp" system cannot truly reflect the player's strength. The "pp" system should directly read the player's mouse trajectory and click time, rather than just limited to checking the player's board performance. The "pp" system sometimes overestimates the player's level or underestimates it because it does not consider everything comprehensively; The "pp" system should output the same performance score when the player's performance level remains unchanged (with the same luck). Now, I can earn 100PP for a single beatmap, but when looking up those very difficult beatmaps, the top one on the ranking can only earn 7PP. Obviously, I cannot be 14 times stronger than those experts, can I? And if I "miss" in some insignificant or simple parts of Beatmap, I may lose a lot of PP, which is unreasonable. This may be difficult, but I believe that performance scores that cannot reflect the true performance of players are even worse. I hope OSU can update the calculation method of "PP" in future versions

Others:

-When I turn on "Classis", the accuracy obtained on the slider does not increase by PP. However, when I turn off "Classis", the accuracy obtained on the slider does not increase by PP, and if I achieve lower accuracy on top, it will decrease by PP

-Some 'mods' do not increase or decrease PP

-PP seems to be related to combo count, which is not good.

(These are machine translation)

**Comment by bdach** [comment]
> The "pp" system should directly read the player's mouse trajectory and click time, rather than just limited to checking the player's board performance

At the scale osu! runs this is infeasible. For the pp system to work, pp values must be recomputable server-side whenever rebalances happen. This would be prohibitively expensive to achieve if replay playback was required for all scores ever set.

> When I turn on "Classis", the accuracy obtained on the slider does not increase by PP. However, when I turn off "Classis", the accuracy obtained on the slider does not increase by PP, and if I achieve lower accuracy on top, it will decrease by PP

I don't understand this part.

> Some 'mods' do not increase or decrease PP

This is known and will eventually be addressed.

> PP seems to be related to combo count, which is not good.

There is pending work on combo scaling removal (https://github.com/ppy/osu/pull/16280).

**Comment by Rekunan** [comment]
All of these issues are already known by the pp devs, I recommend you check out this list and/or join the pp dev server:
https://twitter.com/Natelytlelytle/status/1570486304621854722
https://discord.gg/aqPCnXu
